src/bin/cargo/cli.rs

```diff
@@ -595,6 +606,11 @@ See '<cyan,bold>cargo help</> <cyan><<command>></>' for more information on a sp
                 .value_name("WHEN")
                 .global(true),
         )
+        .arg(
+            opt("warnings", "Warning behavior")
+                .value_parser(["error", "warn", "ignore"])
```

Comment:
You can use [`PossibleValuesParser`](https://docs.rs/clap/latest/clap/builder/struct.PossibleValuesParser.html#method.possible_values) with `TypedValueParser::map` to make this work with the native type, rather than returning a string, centralizing the processing